File: Lv. 2/카펫.py
def solution(brown, yellow):
    if yellow == 1:
        return [3, 3]
    
    for i in range(1, yellow):
        if yellow % i == 0:
            if (i + 2) * (yellow / i + 2) == brown + yellow:
                return [yellow / i + 2, i + 2]

File: Programmers/Lv. 2/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    
    for (int i = 1; i <= int(yellow / 2); i++)
    {
        if ((yellow % i) == 0)
        {
            int w = int(yellow / i);
            int h = i;
            
            if ((int((w + 2) * (h + 2)) - yellow) == brown)
            {
                return {w+2, h+2};
            }
        }
    }
    
    return {3, 3};
}